main.py

```diff
@@ -0,0 +1,99 @@
+import streamlit as sl
+from transformers import pipeline, set_seed
+
+
+def is_int(str):
+    try:
+        int(str)
+        return True
+    except ValueError:
+        return False
+
+
+@sl.cache(suppress_st_warning=True, allow_output_mutation=True)
+def init():
+    return pipeline('text-generation', model='gpt2')
+
+
+GENERATOR = init()
+
+
+def generate_text(input_text, seed, max_len, num_return_seq):
+    if is_int(seed):
+        seed = abs(int(seed))
+        if seed > 2 ** 32 - 1:
+            set_seed(42)
+        else:
+            set_seed(seed)
+    else:
+        set_seed(42)
```

Comment:
It is the most complex logic for setting seed I've ever seen 🤯
There is no reason to do it, especially with hard-coded values (`42`). I suggest you leverage all checks to `set_seed` function and set the default value in function arguments.